### PR TITLE
show goal update confirmation

### LINKED ISSUE #1916: Add confirmation on Goal update
**Additional context**
This was previously implemented on `main` branch thanks to @Shubham-Praj , just has to be ported to `develop`.

**Is your feature request related to a problem? Please describe.**
It's very annoying when you don't get confirmation of your changes. After Goal update the edit modal just closes.

**Describe the solution you'd like**
1. A very brief Toast message 'Goal updated!'
2. A very short (compared to add Goal) sound
Not implemented in `main` - but should be implemented now: if there are no changes, don't show the modal.

**Describe alternatives you've considered**
A confirmation popup. This breaks the flow.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1916

### CODE REVIEW

**Review by tijlleenders — CHANGES_REQUESTED** [review]
This was probably implemented three times now, once in main, once by Nicholas, and then again here...  

Anyway, it looks good.  
Please check why the hints toggle is not being persisted. It is probably an unrelated issue. If so, please file a bug for `develop` and this can be merged.

**Comment by vinaybadgujar102** [comment]
The hint toggle thing is unrelated, Created a new issue for it.

**Review by tijlleenders — APPROVED** [review]
Great!

**Review by Tushar-4781 — COMMENTED** [review]
@vinaybadgujar102 Can you explain a bit what exactly happening in code ?

**Comment by vinaybadgujar102** [comment]
@Tushar-4781, When the updateGoal function is called to modify a goal, we first generate hashes for both the original goal and the updated goal object passed to the function. We then compare these hashes to detect any differences. If differences are detected, we update the goal in the database and display a confirmation toast; otherwise, we simply return.

File: src/hooks/useGoalActions.tsx [hunk 1]
```diff
@@ -67,9 +72,23 @@ const useGoalActions = () => {
       }
       suggestChanges(rootGoal, goal, subGoalsHistory.length);
     } else {
-      await modifyGoal(goal.id, goal, [...ancestors, goal.id], hints);
+      const originalGoal = await getGoal(goal.id);
+      if (originalGoal) {
```
Comment (Tushar-4781) [inline src/hooks/useGoalActions.tsx:76]: club them up -> ( originalGoal && hashObject(originalGoal) !== hashObject(goal))

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

File: src/hooks/useGoalActions.tsx [hunk 2]
```diff
@@ -67,9 +74,21 @@ const useGoalActions = () => {
       }
       suggestChanges(rootGoal, goal, subGoalsHistory.length);
     } else {
-      await modifyGoal(goal.id, goal, [...ancestors, goal.id], hints);
+      const originalGoal = activeGoal;
```
Comment (Tushar-4781) [inline src/hooks/useGoalActions.tsx:77]: no need to duplicate this goal.

File: src/hooks/useGoalActions.tsx [hunk 3]
```diff
@@ -67,9 +74,21 @@ const useGoalActions = () => {
...
+      if (originalGoal && hashObject(originalGoal) !== hashObject(goal)) {
```
Comment (Tushar-4781) [inline src/hooks/useGoalActions.tsx:78]: use activeGoal directly just add a comment that we are comparing the old and updated version.

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

File: src/hooks/useGoalActions.tsx [hunk 4]
```diff
@@ -59,17 +63,31 @@ const useGoalActions = () => {
...
       suggestChanges(rootGoal, goal, subGoalsHistory.length);
-    } else {
+    } else if (activeGoal && hashObject(activeGoal) !== hashObject(goal)) {
+      // Comparing hashes of the old (activeGoal) and updated (goal) versions to check if the goal has changed
+      isGoalChanged = true;
       await modifyGoal(goal.id, goal, [...ancestors, goal.id], hints);
     }
-    setLastAction("goalUpdated");
+
+    if (isGoalChanged) {
```
Comment (Tushar-4781) [inline src/hooks/useGoalActions.tsx:82]: i think you can just move this entire logic in the above block right ?

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**